conans/client/loader.py

```diff
@@ -433,6 +434,10 @@ def _parse_conanfile(conan_file_path):
     current_dir = os.path.dirname(conan_file_path)
     sys.path.insert(0, current_dir)
     try:
+        version_txt = get_required_conan_version_without_loading(conan_file_path)
```

Comment:
I don't love it that we are running a regex against potentially long conanfiles just for this corner case. This will also validate 2 times the same thing, later it is evaluated again.

When we first discussed this issue, you suggested that this would be only checked on ``ImportError`` exceptions, not always, that seemed a bit better.

In any case, if we add this, I would expect it to be only 1.X and drop it in 2.0, is that the plan?

Reply:
But in 2.0 there will be the same issue, right? Why introduce it to 1.X then? Why not just drop this?

Reply:
Yes, that is true, but way more unlikely. We have hit this issue because of the massive migration of imports to new ``conan.tools`` ones, but didn't have this issue in the last 4 years, so it seems that we wouldn't have added it if it wasn't for the migration. In any case, if we are doing it only on import error, I am fine keeping it in 2.0.